### PR TITLE
Fix up forgotten class aliasing issues and typehinting.

### PR DESCRIPTION
**Pull request by dereuromark**
Follow up on https://github.com/cakephp/cakephp/pull/9423

There was a lot of red code and yellow underlining displayed in the IDE.

### CODE REVIEW

File: tests/TestCase/Network/Http/Adapter/StreamTest.php [hunk 1]
```diff
@@ -13,8 +13,8 @@
  */
 namespace Cake\Test\TestCase\Network\Http\Adapter;
 
-use Cake\Network\Http\Adapter\Stream;
-use Cake\Network\Http\Request;
+use Cake\Http\Client\Adapter\Stream;
+use Cake\Http\Client\Request;
```
Comment (markstory) [inline tests/TestCase/Network/Http/Adapter/StreamTest.php]: I had originally left these alone to ensure backwards compatibility with earlier 3.x was retained.

Comment (dereuromark) [inline tests/TestCase/Network/Http/Adapter/StreamTest.php]: I see. So you want to keep them?

Comment (dereuromark) [inline tests/TestCase/Network/Http/Adapter/StreamTest.php]: Currently they display red "Undefined namespace errors" in the IDE.

Comment (jeremyharris) [inline tests/TestCase/Network/Http/Adapter/StreamTest.php]: Can these IDE errors be solved by declaring the namespace above all the `class_alias` files? I tested using:

**src/Network/Http/Request.php**
```php
<?php
namespace Cake\Network\Http;

// @deprecated Backwards compatibility with earler 3.x versions.
class_alias('Cake\Http\Client\Request', 'Cake\Network\Http\Request');
```

In PHPStorm, it won't show the red marks for `Cake\Network\Http\Request` now, and loading that class still loads the `Cake\Http\Client\Request`.

Comment (dereuromark) [inline tests/TestCase/Network/Http/Adapter/StreamTest.php]: Yeah for Projekts they should be Red probably. But Core itself should at some point not internally use deprecations.

**Pull request merged**